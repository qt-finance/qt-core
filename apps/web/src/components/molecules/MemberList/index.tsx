import React from 'react';

import ToggleButton from 'components/atoms/ToggleButton';
import List from 'components/atoms/List';

import { useMember } from 'models/member';

const MemberList: React.FC = () => {
	const [{ staffs }, memberModel] = useMember();

	return (
		<div>
			<ToggleButton onOpen={memberModel.fetchData} onClose={memberModel.reset} />
			<List items={Object.keys(staffs).map(key => ({ key, value: staffs[key].name }))} />
		</div>
	);
};

export default MemberList;
