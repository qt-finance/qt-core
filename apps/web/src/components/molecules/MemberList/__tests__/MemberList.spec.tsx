import React from 'react';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './MemberList.stories';

const testCases = Object.values(composeStories(stories)).map(Story => [
	// The ! is necessary in Typescript only, as the property is part of a partial type
	Story.storyName!, // eslint-disable-line @typescript-eslint/no-non-null-assertion
	Story,
]);
// Batch snapshot testing
test.each(testCases)('Renders %s story', async (_storyName, Story) => {
	const tree = await render(<Story />);
	expect(tree.baseElement).toMatchSnapshot();
});
