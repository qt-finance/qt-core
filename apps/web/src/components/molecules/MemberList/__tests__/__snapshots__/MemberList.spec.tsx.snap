// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Interactive story 1`] = `
<body>
  <div>
    <div>
      <button
        class="button"
        type="button"
      >
        Close
      </button>
      <ul />
    </div>
  </div>
</body>
`;

exports[`Renders WithDefaultMember story 1`] = `
<body>
  <div>
    <div>
      <button
        class="button"
        type="button"
      >
        Close
      </button>
      <ul>
        <li
          data-testid="Jessy"
        >
          Lee
        </li>
      </ul>
    </div>
  </div>
</body>
`;
